api/src/main/java/org/commonjava/indy/content/StoragePathCalculator.java

```diff
@@ -25,4 +25,6 @@
 public interface StoragePathCalculator
 {
     String calculateStoragePath( StoreKey storeKey, String path );
+
+    String calculateStoragePath( StoreKey storeKey, String path, boolean asDirectory);
```

Comment:
I'm just thinking about the parameter name here, and the associated attribute constant. For NPM, we need to avoid interpreting directory paths in a certain way, so the asDirectory name works. But I'm wondering if we want to think of this as a "Raw View" instead, in case we end up implementing package support for something where specific file paths (non-directory) are reinterpreted (instead of just specific directories, as with NPM).

So, I might use `Location.VIEW_RAW` and `boolean rawView` in this parameter...it's stylistic, I know, but I think it might make the code easier to understand for other package types.